daffodil-tdml-lib/src/test/scala/org/apache/daffodil/tdml/ScalaxbTests.scala

```diff
@@ -0,0 +1,23 @@
+package org.apache.daffodil.tdml
+
+import org.apache.daffodil.tdml.scalaxb.TestSuite
+
+import org.junit.Assert._
+import org.junit.Test
+
+class ScalaxbTests {
+
+  @Test def testReading(): Unit = {
+    val testSuite =
+      _root_.scalaxb.fromXML[TestSuite](
+        scala.xml.XML.load(
+          getClass
+            .getClassLoader()
+            .getResourceAsStream("test-suite/ibm-contributed/dpaext1-2.tdml"),
+        ),
+      )
+
+    assertNotNull(testSuite)
+    assertEquals(Some("dpaext"), testSuite.suiteName)
+  }
+}
```

Comment:
While trying to update the tests, I tried to parse IPv4.tdml from another module's tests and discovered scalaxb does not properly ignore XML comments when parsing, failing to parse. This is likely a bug in the generated parsing code of scalaxb, and I'll file an issue, but it may be a blocker for us. What do you think if this is the case @stevedlawrence @mbeckerle?

Reply:
We have various tools in daffodil-misc/xml/XMLUtils.scala that strip out comments, strip out processing instructions, etc. 
You may need to do that pass first in order to be able to read in TDML.

This is a really common thing in XML processing that people don't accommodate comments, which are part of the XML Infoset. As are EntityRefs, etc. Somewhere in the daffodil tests we have a DFDL schema that has a comment everwhere that a comment is allowed. To make sure we tolerate them in all the places they can appear. 

Are you parsing TDML, or reading and writing it back?

Reading one can strip out this stuff, but if you are rewriting the file this stuff has to be preserved. TDML is not an object model. Ex: reading and writing this:
```
<str><![CDATA[big long string with
significant whitespace
inside it.
]]></str>
```
And getting it to round trip with that CDATA bracketing in it, all whitespace preserved, etc. .... that's going to be tricky.

Reply:
> but if you are rewriting the file this stuff has to be preserved

I don't think jaxb/scalaxb has ways to preserve comments/cdata etc. It's all converted to Java objects that don't retain that information.

I think this just needs to be a limitation for people wanting to use jaxb/scalaxb, they need to strip comments prior to unmarshalling, and roundtripping is not guaranteed when marshalling.

Reply:
@arosien what is the use case here. I had thought the purpose of this scalaxb stuff was to enable adding TDML support to the VSCode extension, and that would mean both reading and writing out TDML.

Reply:
The use case for the extension is to enable reading and writing of **generated** TDML files, so the comments aren't an immediate issue for us. We don't expect users to have or use non-generated TDML files in order to use the features in the extension.

Reply:
Can VS Code extensions use Java/Scala libraries? I thought everything was typescript?

Reply:
The frontend is typescript. The backend that communicates with Daffodil is Scala.

Reply:
> The use case for the extension is to enable reading and writing of generated TDML files, so the comments aren't an immediate issue for us. We don't expect users to have or use non-generated TDML files in order to use the features in the extension.

I don't quite think this is true. We happen to most often handle completely generated TDML files, but it should be the case that someone with a pre-existing non-generated TDML file would want to add a generated test case to the file, in which case the app using this lib should either raise an error if there are comments, or warn that we're going to trash them.

> @arosien what is the use case here. I had thought the purpose of this scalaxb stuff was to enable adding TDML support to the VSCode extension, and that would mean both reading and writing out TDML.

It's both reading and writing in the daffodil-vscode project, but I don't think that matters for the changes in daffodil proper. We can document the limitation with comments somewhere and downstream projects can choose to raise an error when encountering comments, filter them out, etc.

Reply:
> Can VS Code extensions use Java/Scala libraries? I thought everything was typescript?

> The frontend is typescript. The backend that communicates with Daffodil is Scala.

Oh, I forgot, this is my favorite part... We currently launch the daffodil debugger to create and append TDML files, which is IMO rather excessive, given that we could alternatively read and write XML files from the typescript extension itself. (There are a number of typescript-based XSD XML parser generator libraries out there.) But we can perhaps discuss that in a different forum.

Reply:
> We currently launch the daffodil debugger to create and append TDML files, which is IMO rather excessive, given that we could alternatively read and write XML files from the typescript extension itself.

Yeah, a pure typescript implementation has some clear advantages. That might be a better approach.

Another thought, there have been some discussions about an alternative DFDL syntax since there's a lot of dislike of XML. Maybe it's also worth considering a different test format? Maybe json (or something similar) would be sufficient, which I believe typescript has much better support for. That probably requires an even longer discussion and is maybe a more long term thing, if even done, but possibly something worth considering.

Reply:
> it should be the case that someone with a pre-existing non-generated TDML file would want to add a generated test case to the file, in which case the app using this lib should either raise an error if there are comments, or warn that we're going to trash them.

Agreed that we should, but the immediate use case is for generated files. The hope is that people with custom non-generated files that don't work out of the box create an issue and we can deal with the edge cases that way. I think having some sort of message/warning if comments (or anything else that we can't currently handle) are detected is a really good starting point that we're not currently doing - a ticket to track this would probably be useful.

> We currently launch the daffodil debugger to create and append TDML files

This isn't quite as true as it was before. With the GUI implemented, we've integrated the TDML generate/append more into the actual DFDL parse. By default, a TDML file is generated for every parse - it's not a separate action anymore. The generate/append actions work off of the last parse that was ran instead of requiring the user to do another parse.

So, when the user hits the generate/append button, it's all handled on the typescript side now. Even though that's not the complete picture, it's a step towards decoupling the TDML from the backend if that's a direction we eventually move to... Although, I don't see how we can do that as a TDML test case (or at least for how we're using it) is implicitly tied to a DFDL parse. Maybe there's a path to take if we add a way to have the backend send information back to the frontend after a parse?

Reply:
> It's both reading and writing in the [daffodil-vscode](https://issues.apache.org/jira/browse/DAFFODIL-vscode) project, but I don't think that matters for the changes in daffodil proper. We can document the limitation with comments somewhere and downstream projects can choose to raise an error when encountering comments, filter them out, etc.

So when scalaxb writes out an infoset, strings in the infoset are likely to contain "<", ">", "&", etc. So they need XML escapifying when output. I would assume this would happen automatically. 

But note that if you read in TDML and an infoset is embedded and uses `<![CDATA[...]]>` stuff, that it will write out as the string contents (without the CDATA bracketing) but escapified. 

I'm ok with that. It's just fragile if XML tooling decides it is ok to mess with the whitespace of the TDML file. E.x., if you output the XML back to a file by pretty printing it, then data might be corrupted by whitespace changes. 

This is just a really painful area of experience with XML. It's just not a great data language because it was designed to be a human authored markup language. Hence, the whitespace insensitivity, line-ending changes, etc.

Reply:
> So, when the user hits the generate/append button, it's all handled on the typescript side now.

ah, i forgot about that, thanks @michael-hoke